### PR TITLE
Make code formatting script as a CMake module

### PR DESCRIPTION
**Pull request by egordon**
Solve issue #232 

Added CMake module for Clang Format based on the one from Chimera. Can specify requested clang-format version.

***

**Before creating a pull request**

- [ n/a ] Document new methods and classes
- [ x ] Format code with `make format`

**Before merging a pull request**

- [ n/a ] Set version target by selecting a milestone on the right side
- [ n/a ] Summarize this change in `CHANGELOG.md`
- [ n/a ] Add unit test(s) for this change

### CODE REVIEW

File: cmake/ClangFormat.cmake [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  find_program(CLANG_FORMAT_EXECUTABLE NAMES clang-format-${clang_version})
+  if(NOT CLANG_FORMAT_EXECUTABLE)
+    message(STATUS "Looking for clang-format-${clang_version} - NOT found,"
+      "please install to enable automatic code formatting."
+    )
+    return()
+  endif()
+
+  message(STATUS "Found clang-format-${clang_version}.")
+endfunction()
```
Comment (jslee02) [inline cmake/ClangFormat.cmake]: 7Suggestion: `clang_format_setup` and `clang_format_setup` can be merged by using [`cmake_parse_arguments`](https://cmake.org/cmake/help/v3.12/command/cmake_parse_arguments.html).

File: cmake/ClangFormat.cmake [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    message(STATUS "Looking for clang-format - NOT found, please install "
+      "clang-format to enable automatic code formatting."
+    )
+    return()
+  endif()
+
+  message(STATUS "Found clang-format.")
+endfunction()
+
+function(clang_format_setup_version clang_version)
```
Comment (jslee02) [inline cmake/ClangFormat.cmake]: Nit: Add the line separator for consistency.

File: CMakeLists.txt [hunk 3]
```diff
@@ -144,6 +115,7 @@ find_package(ompl REQUIRED)
 # TREAT_WARNINGS_AS_ERRORS.
 include(CompilerSettings)
 
+
```
Comment (jslee02) [inline CMakeLists.txt]: Nit: Remove whitespace

File: cmake/ClangFormat.cmake [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  add_custom_target(check-format
+    COMMAND ${CMAKE_COMMAND} -E echo "Checking ${formatting_files_length} files..."
+    COMMAND ${CMAKE_SOURCE_DIR}/tools/check_format.sh ${CLANG_FORMAT_EXECUTABLE} ${formatting_files}
+    COMMAND ${CMAKE_COMMAND} -E echo "Done."
+    DEPENDS ${CLANG_FORMAT_EXECUTABLE}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+endfunction()
```
Comment (jslee02) [inline cmake/ClangFormat.cmake]: This function wouldn't work when `formatting_files` is empty. Consider having [this logic](https://github.com/personalrobotics/aikido/pull/509/files#diff-af3b638bc2a3e6c650974192a53c7291L314).

File: cmake/ClangFormat.cmake [hunk 5]
```diff
@@ -0,0 +1,90 @@
+function(clang_format_setup)
+  cmake_parse_arguments(
+        CF_ARG # prefix
```
Comment (brianhou) [inline cmake/ClangFormat.cmake]: It seems a little inconsistent to me that the prefix is `CF_ARG` but the exported executable is `CLANG_FORMAT_EXECUTABLE`. I'm having a hard time finding useful examples on the internet, does this reasonably follow CMake conventions @jslee02?

Comment (jslee02) [reply cmake/ClangFormat.cmake]: CMake is not a good tool to find a good convention of using it. See this [list](https://github.com/onqtam/awesome-cmake). It's horrific. :scream:

Back to the question, it seems people uses lower caess for the prefix (e.g., here and here). Also, there is tendency of using upper cases for global variables and lower cases for local variables. By the way, `CLANG_FORMAT_EXECUTABLE` is set by `find_program` not by `cmake_parse_arguments`.

**Pull request merged**